### PR TITLE
Kanban/Board View

### PR DESCRIPTION
**Pull request by hieugao**
I implemented some of the basic (front-end only) for Board View, it's very primitive at this point and also I have zero knowledge about the back-end. So if anyone is interested please help me to finish the feature.

### Current state

https://user-images.githubusercontent.com/13729196/153836842-e5a08131-48b0-4d31-8b2e-762f606ba202.mp4

### Tasks
- [x] Moving (reordering) Column and Card - _the heaviest work is already done thanks to [`boardview`](https://pub.dev/packages/boardview)_
- [x] Add new Column and Card
- [ ] Edit Column and Card
- [ ] Delete Column and Card
- [ ] Add more Card's properties beside `title` like `tags`, `image`, etc
- [ ] Scrolling using mouse
- [ ] Focus on new Column when it is out of sight - _maybe [this](https://pub.dev/packages/ensure_visible_when_focused) could help_
- [ ] Polish the UI to match [the design](https://images.squarespace-cdn.com/content/v1/617f6f16b877c06711e87373/1636772749116-RB01WX0STUQOATFOS3QT/project-management-task), check out [AppFlowy: Brand](https://appflowy.gitbook.io/docs/design-documentation/ui-design-system-documentation) for more details (including enhancement if the design has any flaws) - _I'm currently focusing on the logic (function) side first_

Backend:

- [ ] Glue the front-end code with the back-end (or the app in general)

### Contributing
- __Issue or Discussion__ - you could open it side this PR
- __PR__ - I think you should open it to:
  - https://github.com/hieugao/appflowy/tree/feat/kanban-view
  - https://github.com/hieugao/boardview
- Or __DM__ me on AppFlowy Discord Server if you have any questions

### Big Thanks To
_(I don't know if tagging them is appropriate so I removed @ sign)_

- jakebonk - for [`boardview`](https://github.com/jakebonk/FlutterBoardView)
- invoiceninja - for the [reference](https://github.com/invoiceninja/admin-portal/search?q=kanban)

### CODE REVIEW

File: frontend/app_flowy/lib/workspace/presentation/kanban_view/board_page.dart [hunk 1]
```diff
@@ -0,0 +1,162 @@
+// import 'package:boardview/board_item.dart';
+// import 'package:boardview/board_list.dart';
```
Comment (appflowy) [inline frontend/app_flowy/lib/workspace/presentation/kanban_view/board_page.dart:2]: I think you should not commit the comment-out codes.

**Comment by appflowy** [comment]
Hi,@hieugao, I will build up the generic data layer (In the back-end) to store the Kanban/Board data.

**Comment by hieugao** [comment]
@appflowy, yo I'm glad hearing that.

Maybe after this PR get merged we could write a guide on "How a Front-end developer can create a new feature without knowing much about the Back-end".

File: frontend/app_flowy/pubspec.yaml [hunk 2]
```diff
@@ -45,6 +45,10 @@ dependencies:
       ref: dbc309f5e382963fa0122409a0aaf8e1417d51c1
 
   #  third party packages
+  boardview:
+    git:
+      url: git://github.com/hieugao/boardview.git
```
Comment (Hari-07) [inline frontend/app_flowy/pubspec.yaml:50]: Any issues that prevent using the pub.dev version?

Comment (appflowy) [reply frontend/app_flowy/pubspec.yaml:50]: I think push to pub.dev would be better.

Comment (Hari-07) [reply frontend/app_flowy/pubspec.yaml:50]: I'm against using a package that's not in active development. Better to implement it ourselves in that case

Comment (hieugao) [reply frontend/app_flowy/pubspec.yaml:50]: That's why I said it will be included into AppFlowy.

**Comment by rileyhawk1417** [comment]
@hieugao tried viewing this PR locally failing to clone package using flutter get

**Comment by appflowy** [comment]
@hieugao  I have finished the basic Grid. The data structs of Grid can share with Kanban. Could you mind migrating the code into AppFlowy? I want to modify it.

**Comment by hieugao** [comment]
Hi @appflowy, so sorry for putting it into this state, I got quite busy nowadays, so I think I'm going to close this PR and let's you implement the feature.

**Comment by appflowy** [comment]
Ok, thank you!

**Pull request closed without merging**